src/runtime/virtcontainers/sandbox.go

```diff
@@ -1657,9 +1657,19 @@ func (s *Sandbox) StatsContainer(ctx context.Context, containerID string) (Conta
 // Stats returns the stats of a running sandbox
 func (s *Sandbox) Stats(ctx context.Context) (SandboxStats, error) {
 
-	metrics, err := s.sandboxController.Stat()
-	if err != nil {
-		return SandboxStats{}, err
+	var metrics interface{}
+	var err error
+	if !s.config.SandboxCgroupOnly {
+
+		metrics, err = s.overheadController.Stat()
+		if err != nil {
```

Comment:
You can check the err after `if !s.config.SandboxCgroupOnly { } else {}`, then you can save one err check.